File: app-java/src/test/java/tech/thdev/java_udemy_sample/NullTest.java
package tech.thdev.java_udemy_sample;

import android.support.annotation.Nullable;

import org.junit.Test;

/**
 * Created by tae-hwan on 10/6/16.
 */

public class NullTest {

    @Test
    public void testNullable() throws InterruptedException {
        String name = "Name";
        System.out.print(getLength(name));
    }

    private int getLength(@Nullable String name) {
        return name.length();
    }

































    @Test
    public void testNull() throws InterruptedException {
        String name = null;
        System.out.print(getLength(name));
    }
}

File: app-kotlin/src/test/java/tech/thdev/kotlin_udemy_sample/NullTest.kt
package tech.thdev.kotlin_udemy_sample

import org.junit.Test

/**
 * Created by tae-hwan on 10/6/16.
 */
class NullTest {

    @Test
    @Throws(InterruptedException::class)
    fun testNullable() {
        val name = "Name"
        print(getLength(name))
    }

    private fun getLength(name: String?): Int {
        return name!!.length
    }


































    @Test
    @Throws(InterruptedException::class)
    fun testNull() {
        val name = null
        print(getLength(name))
    }
}